### PR TITLE
Create and save csv-files on samba share

### PR DESCRIPTION
**Pull request by cheezi0747**
Also includes fixes to problems that occured during testing of the samba-integration locally.
Tests for the samba integration is still to be implemented since we have problems with testcontainers not working as it should

### CODE REVIEW

File: src/main/resources/db/migration/V1_0__create_initial_schema.sql [hunk 1]
```diff
@@ -23,18 +23,25 @@ create table department
     primary key (id)
 ) engine = InnoDB;
 
+create table recipient
+(
+    id          bigint not null auto_increment,
+    adress      varchar(255),
```
Comment (foobar-1337) [inline src/main/resources/db/migration/V1_0__create_initial_schema.sql:29]: address ?

File: src/main/resources/db/migration/V1_0__create_initial_schema.sql [hunk 2]
```diff
@@ -23,18 +23,25 @@ create table department
...
+    city        varchar(255),
+    co          varchar(255),
```
Comment (foobar-1337) [inline src/main/resources/db/migration/V1_0__create_initial_schema.sql:31]: In english "co" is an abbreviation of "company".
Maybe use "care_of" instead?

File: src/main/java/se/sundsvall/snailmail/integration/samba/SambaIntegration.java [hunk 3]
```diff
@@ -27,68 +32,113 @@
...
+				final var name = request.getRecipient().getGivenName() + " " + request.getRecipient().getLastName();
+				final var careOf = request.getRecipient().getCo();
+				final var address = request.getRecipient().getAdress();
+				final var postalCode = request.getRecipient().getPostalCode();
+				final var city = request.getRecipient().getCity();
+
+				printWriter.printf("%s,%s,%s,%s,%s%n", name, careOf, address, postalCode, city);
 			}
+		} catch (final IOException e) {
+			throw Problem.valueOf(Status.INTERNAL_SERVER_ERROR, "Failed to write  to Samba share");
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/snailmail/integration/samba/SambaIntegration.java:103]: Might want to log this. Same with other catches in this class.

File: src/main/resources/db/migration/V1_0__create_initial_schema.sql
Comment (foobar-1337) [inline src/main/resources/db/migration/V1_0__create_initial_schema.sql:1]: DepartmentRepository has a findByName()-method. Should we have an index for department.name?

File: src/main/java/se/sundsvall/snailmail/integration/db/model/Attachment.java
Comment (foobar-1337) [inline src/main/java/se/sundsvall/snailmail/integration/db/model/Attachment.java:1]: I personally like the convention of adding suffixes on classnames that describes the purpose.
(I.e.. This class would be AttachmentEntity.)

File: src/main/java/se/sundsvall/snailmail/api/model/SendSnailMailRequest.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ public static class Attachment {
 		@NotBlank
 		private String content;
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/snailmail/api/model/SendSnailMailRequest.java:53]: Is this base64-content?

There is a ValidBase64-annotation available in dept44-common-validators (if you want to add validation for this?)

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**